src/components/lib/pci/pci.c

```diff
@@ -57,16 +55,13 @@ pci_scan(struct pci_dev *devices, int sz)
 					devices[dev_num].bar[k].paddr = devices[dev_num].bar[j].raw & 0xFFFFFFF0;
 
 				}
-				
 				dev_num++;
-				if(dev_num >= sz || dev_num >= PCI_DEVICE_MAX) {
-					return 0;
-				}
+				assert(dev_num < PCI_TOTAL_MAX);
```

Comment:
Comment here for what error happened. If people hit this, they need to know how to fix it.